state-commitments/optimized/history_commitment.go

```diff
@@ -50,18 +77,43 @@ func NewCommitter() *HistoryCommitter {
 	}
```

Comment:
In general, it seems unclear to me why HistoryCommitter is a struct or what's stored in it. Is it expected to be reused across different trees? There's a bunch of arguments to computeNodeHash, and some of them are just about the structure of the tree, e.g. numRealLeaves. Could those be stored in this struct? If not, maybe we should have a TreeInfo struct with all that data and easy helper functions for paddingStartIndexAtLevel and such?

Reply:
historyCommitter is now encapsulated in the history package and callers only interact with package-level functions. Those functions do still use the history committer for sharing some state during recursion just to keep from having to pass so many arguments or a pointer to the struct through recursive function calls.